app/components/Intercept_Components/InterceptTextBox.tsx

```diff
@@ -2,12 +2,19 @@ import * as React from "react";
 import {RequestHeaderList} from "./RequestHeaderList";
 
 export const InterceptTextBox = props => {
+  const isObject = (val:any) => {
+    if (val === null) { return false;}
+    return ( (typeof val === 'function') || (typeof val === 'object') );
+  }
+  const requestId = props.rowProps.checkbox.requestId
+  console.log("Re-rendered")
+  console.log("responseData::", props.responseData[requestId])
   const defaultResponseText = "";
   const defaultStatusCode = "200";
   const defaultContentType = "application/json";
-  const responseTextValue = props.responseText[props.rowProps.checkbox.requestId] || defaultResponseText;
-  const statusCodeValue = props.statusCodes[props.rowProps.checkbox.requestId] || defaultStatusCode;
-  const contentTypeValue = props.contentType[props.rowProps.checkbox.requestId] || defaultContentType;
+  const responseTextValue = isObject(props.responseText[requestId]) ? JSON.stringify(props.responseText[requestId]) : props.responseText[requestId] || defaultResponseText;
```

Comment:
Ideally, it's more cleaner and easy to follow if the type of data is consistent: either always ensure it's an object, or ensure it's always a string (at this function call site). let the caller do the processing. That way, your main interception code doesn't need to handle all kinds of complex marshalling logic.